### DISCUSSION TITLE
Equivalent to Microsoft.Toolkit.Uwp.UI.Media.Geometry.CanvasPathGeometry in the Toolkit, Labs, elsewhere?

### DISCUSSION THREAD

**Discussion by mendedwraith** [CommunityToolkit/Windows#564] [Q&A]
I'm looking for the helper methods related to `Microsoft.Toolkit.Uwp.UI.Media.Geometry.CanvasPathGeometry` that are compatible with WinUI 3. Namely these methods:
```
public static Microsoft.Graphics.Canvas.Geometry.CanvasStrokeStyle CreateStrokeStyle (string styleData);

public static ICanvasStroke CreateStroke(ICanvasResourceCreator resourceCreator, string strokeData);

// CanvasPathGeometry
public static CanvasGeometry CreateSquircle(ICanvasResourceCreator resourceCreator, float x, float y, float width, float height, float radiusX, float radiusY);
```

This [post](https://learn.microsoft.com/en-us/windows/communitytoolkit/parsers/canvaspathgeometry) is from some time ago and I found my way to CommunityToolkit/Windows. As far as I can tell `Microsoft.Toolkit.Uwp.UI.Media` isn't compatible with WinUI 3 and I haven't been able to find a suitable alternative.

**Comment by optsing** [comment]
Apparently you need the Win2D package (Microsoft.Graphics.Win2D for WinUI), but it seems that the required figure is not there. As a solution, you can copy the code from the obsolete [WindowsCommunityToolkit](https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/main/Microsoft.Toolkit.Uwp.UI.Media/Geometry/CanvasPathBuilderExtensions.cs#L353).
